core-common/src/main/java/org/apache/kylin/common/KylinConfigBase.java

```diff
@@ -3088,4 +3088,12 @@ public String getKerberosJaasConfPath() {
     public String getKerberosPrincipal() {
         return getOptional("kylin.kerberos.principal");
     }
+
+    public int getRepartitionNumAfterEncode(){
+        return Integer.valueOf(getOptional("kylin.engine.spark.dataset.repartition.num.after.encode", "0"));
```

Comment:
it's better change the key to 'kylin.engine.spark.dataset.repartition.num.after.encoding'